Provisioning: Document allowed_git_urls setting (#129670)
grafana/grafana: docs/sources/as-code/observability-as-code/git-sync/git-sync-setup/set-up-before.md

- [Role and resource permissions](#resource-and-role-permissions)
- For further details on how Git Sync operates, refer to the [key concepts](https://grafana.com/docs/grafana/<GRAFANA_VERSION>/as-code/observability-as-code/git-sync/key-concepts)

## Allow internal or private Git servers

{{< admonition type="note" >}}

> This setting is available starting in Grafana v13.0.0 and applies to self-managed Grafana (OSS and Enterprise) only. It's not configurable in Grafana Cloud.

Author: This flag is available starting:

- `v13.0.4`
- `v13.1.1`

and from `v13.2.x` onwards

1. Save the changes to the file and restart Grafana.

## Allow internal or private Git servers

> By default, Git Sync rejects repository URLs whose host resolves to a loopback, private (RFC 1918), link-local, or unspecified address. This protects your Grafana instance against server-side request forgery (SSRF). Public Git servers such as `github.com`, `gitlab.com`, and `bitbucket.org` resolve to public addresses and are always allowed.

Reviewer: ```suggestion
While Public Git servers such as `github.com`, `gitlab.com`, and `bitbucket.org` resolve to public addresses and are always allowed, by default Git Sync rejects repository URLs with a host that resolves to a loopback, a private (RFC 1918), link-local, or an unspecified address. This protects your Grafana instance against server-side request forgery (SSRF). 
```

> This setting applies to self-managed Grafana (OSS and Enterprise). It's not configurable in Grafana Cloud.

Reviewer: I'd move this to the top and as a note!

> If you connect Git Sync to a Git server on a private network — for example, a self-hosted GitHub Enterprise, GitLab, or Bitbucket instance reachable only through an internal address — add its host to the `allowed_git_urls` allowlist:

Reviewer: ```suggestion
If you connect Git Sync to a Git server on a private network such as a self-hosted GitHub Enterprise, GitLab, or Bitbucket instance reachable only through an internal address, add its host to the `allowed_git_urls` allowlist:
```

1. Open your Grafana configuration file, either `grafana.ini` or `custom.ini`.
1. Add each internal Git host to `allowed_git_urls` as a comma-separated list:

   ```ini
   [provisioning]
   allowed_git_urls = git.internal.example.com, ghe.example.com:8443
   ```

1. Save the changes to the file and restart Grafana.

> Each entry can be a hostname, `host:port`, a full URL (only the host is used), a literal IP address, or a CIDR range. Prefer specific hosts or narrow ranges: a broad CIDR such as `10.0.0.0/8` re-exposes the entire private range that SSRF protection blocks. Only add hosts you trust — an allowlisted host receives the configured Git token on every sync, fetch, and push.

Reviewer: ```suggestion
Each entry can be a hostname, `host:port`, a full URL (only the host is used), a literal IP address, or a CIDR range. If possible, use specific hosts or narrow ranges, since a broad CIDR such as `10.0.0.0/8` re-exposes the entire private range that SSRF protection blocks. 

**Only add hosts you trust** because an allowlisted host receives the configured Git token on every sync, fetch, and push.
```
